src/index.js

```diff
@@ -107,24 +107,18 @@ export default function promiseMiddleware(config = {}) {
        *   }
        * }
        */
-      return new Promise((resolve, reject) => {
-        promise.then(
-          (value = null) => {
-            const resolvedAction = getAction(value, false);
-            dispatch(resolvedAction);
-            resolve({ value, action: resolvedAction });
-
-            return;
-          },
-          (reason = null) => {
-            const rejectedAction = getAction(reason, true);
-            dispatch(rejectedAction);
-            reject({ reason, action: rejectedAction });
-
-            return;
-          }
-        );
-      });
+      return promise
+        .then((value = null) => {
+          const resolvedAction = getAction(value, false);
+          dispatch(resolvedAction);
+          return { value, action: resolvedAction };
+        })
+        .catch((reason = null) => {
```

Comment:
This seems dangerous... it could lead to both _fulfilled **and** _rejected being dispatched.

If `dispatch(resolvedAction)` causes something else to throw, like a component... then a reducer could be undoing the state update from the _fulfilled!